### ISSUE TITLE
Not possible to receive DTS core audio or Dolby Digital core audio over SPDIF?

### ISSUE THREAD

**Issue by Aginarikar** [koying/SPMC#829]
Been using USB audio DAC to SPDIF/optical many years with Kodi on Linux/Windows.
As you may know, DTS HD and Dolby Digital Plus tracks contain a **core** audio track which the SPDIF connection can use since it is only an ordinary DTS or Dolby Digital track _(Dolby Digital TrueHD doesn't contain core audio)_.

Moving on to Nvidia Shield 2017 running on Android 7.0 and SPMC both 16.6.0/16.5.5 from standard DTS and Dolby Digital tracks the audio works fine when SPMC set as Passthrough device RAW.

-If "Transcode to AC-3" is selected, DTS HD and Dolby Digital Plus will be converted to Dolby Digital.
-But if "Transcode to AC-3" is unselected, there's either no sound, or just a 2.0 PCM or Dolby Digital track.
I've tried to check if this is because of some limits in the Shield itself, Android 7.0, Kodi or SPMC but haven't found anything. My closest suspicion is that it could be a bug of Android 7.0?

**Comment by btc3111** [comment]
Same issue; the shitty Plex app for shield 7.0 does that crashes way to often and when Kodi 17 RAW passthrough worked on Android 6.0 for the shield it was also capable of using the DTS core. Here's hoping koying is able to implement this as SPMC is by far the most well maintained Android fork since the Kodi devs refuse to reimplement SPDIF passthrough

**Comment by koying** [comment]
What happens if you select RAW and enable DTS-HD in SPMC?

RAW means android handles everything, so whether DTS-CORE is extracted from DTS-HD if the audio output doesn't support it is entirely up to the system

**Comment by btc3111** [comment]
Just tried it. It will play the video without any sound. Also tried IEC because why not and same symptom. That's unfortunate it's system side, will just have to keep using the crappy Plex app as a work around then.

**Comment by koying** [comment]
@btc3111 You also have a USB DAC?
What happens in IEC with DTS-HD disabled? In my "simple" Shield->AVR setup, DTS-CORE is properly extracted...

**Comment by btc3111** [comment]
Yes, I have a USB DAC->SPDIF soundbar. IEC doesn't work no matter what bitstreaming is selected or not. it's just the same issue of no sound;  that's at least via the Turtle Beach branded DAC I own. RAW is the only working option for bitstreaming for myself. I've also checked the Android tv settings numerous times as well. Its really just time to upgrade to an AVR anyway. Thanks for troubleshooting btw

**Comment by txemix** [comment]
I had a similar setup: Shield + Turtle Beach SPDIF adapter, I move to a new receiver but I'm keeping SPDIF because of auto refresh rate not working with HDCP 1.4 TV. (By the way, it's a pity that NVidia cannot provide a simple solution to this mixed HDCP problem, even a trivial option to select HDCP 1.4 would do as some user has shown [here](https://forums.geforce.com/default/topic/901911/shield-tv/auto-refresh-rate-switching-not-working/post/5110689/#5110689) ).

With Android 6.0 and SPMC, IEC worked, and DTS HD provided DTS core to receiver via SPDIF.

Now with Android 7.0, IEC doesn't work with SPDIF output (no sound), so RAW must be used, but cannot get DTS core from HD audio (using passthrough). As per koying explanation I understand this is due to Android.

**Comment by Aginarikar** [comment]
Thanks everyone so far!
